ckan/views/__init__.py

```diff
@@ -156,35 +133,29 @@ def identify_user() -> Optional[Response]:
 def _identify_user_default():
     u'''
     Identifies the user using two methods:
-    a) If they logged into the web interface then repoze.who will
-       set REMOTE_USER.
-    b) For API calls they may set a header with an API key.
+    a) If they logged into the web interface then flask-login will
+       set _user_id into beaker.session.
+    b) For API calls they may set a header with an API token.
     '''
 
-    # environ['REMOTE_USER'] is set by repoze.who if it authenticates a
-    # user's cookie. But repoze.who doesn't check the user (still) exists
-    # in our database - we need to do that here. (Another way would be
-    # with an userid_checker, but that would mean another db access.
-    # See: http://docs.repoze.org/who/1.0/narr.html#module-repoze.who\
-    # .plugins.sql )
+    # beaker_session['_user_id'] is set by flask-login if it authenticates a
+    # user's cookie. But flask-login doesn't check the user (still) exists
+    # in our database - we need to do that here.
     g.user = six.ensure_text(request.environ.get(u'REMOTE_USER', u''))
     if g.user:
         g.userobj = model.User.by_name(g.user)
-
         if g.userobj is None or not g.userobj.is_active():
 
             # This occurs when a user that was still logged in is deleted, or
             # when you are logged in, clean db and then restart (or when you
             # change your username). There is no user object, so even though
-            # repoze thinks you are logged in and your cookie has
+            # flask-login thinks you are logged in and your cookie has
             # ckan_display_name, we need to force user to logout and login
             # again to get the User object.
-
-            ev = request.environ
-            if u'repoze.who.plugins' in ev:
-                pth = getattr(ev[u'repoze.who.plugins'][u'friendlyform'],
-                              u'logout_handler_path')
-                redirect(pth)
+            from flask_login import logout_user
+            logout_user()
+            g.userobj = None
+            g.user = None
```

Comment:
Is this piece of code still necessary? It looks like we are trying to fit a `repoze` logic to `flask-login`.

It says: 
```
#...But flask-login doesn't check the user (still) exists
# in our database - we need to do that here.
```
But we have the `@login_manager.user_loader` that specifically looks for the user in the Database.

Reply:
@pdelboca 
I think we don't need this code anymore, we don't need to invent a way to pass the current logged user, as we do here with  `g.user/g.userobj` as we already have it for free in `current_user` proxy from `flask_login`, and I think we should start using it. 

I started to make these changes step by step.

Please, if you have more suggestions, let me know, or feel free to make a commit :)

Reply:
@TomeCirun we still need `g.user/g.userobj` as is widely used in extensions too. But I think this logic can be further clean. 

Some suggestions:
 - What is the goal of `_identify_user_default()`? Looks like is not identifying the user but rather setting the `g.user/g.userobj` attributes. In this case, I would rename to be more explicit on the function's goal.
 - Since this is CKAN core, I would try to set `g.user/g.userobj` using Flask functions. (Do not rely on `REMOTE_USER` anymore.)

Note:
- REMOTE_USER is also widely used in test, so we may need to mantain it here. However, If needed for tests I would try to either add a specific documented line of code or move the logic to the tests fixtures.

I would try to add an example commit later this week. :)